java/integTest/src/test/java/glide/SharedCommandTests.java

```diff
@@ -2288,11 +2288,81 @@ public void xadd(BaseClient client) {
-         * threshold: 1, exact: true, }), ).toEqual(1); expect(await client.customCommand(["XLEN",
-         * key])).toEqual(1);
-         */
+        // test xtrim to remove 1 element
+        assertEquals(1L, client.xtrim(key, new MaxLen(1)).get());
+        // TODO update test when XLEN is available
+        if (client instanceof RedisClient) {
+            assertEquals(1L, ((RedisClient) client).customCommand(new String[] {"XLEN", key}).get());
+        } else if (client instanceof RedisClusterClient) {
+            assertEquals(
+                    1L,
+                    ((RedisClusterClient) client)
+                            .customCommand(new String[] {"XLEN", key})
+                            .get()
+                            .getSingleValue());
+        }
+
+        // Key does not exist - returns 0
+        assertEquals(0L, client.xtrim(key, new MaxLen(true, 1)).get());
+
+        // Key exists, but it is not a stream
+        assertEquals(OK, client.set(key2, "xtrimtest").get());
+        ExecutionException executionException =
+                assertThrows(ExecutionException.class, () -> client.xtrim(key2, new MinId("0-1")).get());
+        assertTrue(executionException.getCause() instanceof RequestException);
+    }
+
+    @SneakyThrows
+    @ParameterizedTest(autoCloseArguments = false)
+    @MethodSource("getClients")
+    public void xread(BaseClient client) {
+        String key1 = "{key1}";
+        String key2 = key1 + "-" + UUID.randomUUID();
+        String field1 = "f1_";
+        String field2 = "f2_";
+        String field3 = "f3_";
+
+        // setup first entries in streams key1 and key2
+        Map<String, String> timestamp_1_1_map = new LinkedHashMap<>();
+        timestamp_1_1_map.put(field1, field1 + "1");
+        timestamp_1_1_map.put(field3, field3 + "1");
+        String timestamp_1_1 = client.xadd(key1, timestamp_1_1_map).get();
+        assertNotNull(timestamp_1_1);
+
+        String timestamp_2_1 = client.xadd(key2, Map.of(field2, field2 + "1")).get();
+        assertNotNull(timestamp_2_1);
+
+        // setup second entries in streams key1 and key2
+        String timestamp_1_2 = client.xadd(key1, Map.of(field1, field1 + "2")).get();
+        assertNotNull(timestamp_1_2);
+
+        String timestamp_2_2 = client.xadd(key2, Map.of(field2, field2 + "2")).get();
+        assertNotNull(timestamp_2_2);
+
+        // setup third entries in streams key1 and key2
+        Map<String, String> timestamp_1_3_map = new LinkedHashMap<>();
+        timestamp_1_3_map.put(field1, field1 + "3");
+        timestamp_1_3_map.put(field3, field3 + "3");
+        String timestamp_1_3 = client.xadd(key1, timestamp_1_3_map).get();
+        assertNotNull(timestamp_1_3);
+
+        String timestamp_2_3 = client.xadd(key2, Map.of(field2, field2 + "3")).get();
+        assertNotNull(timestamp_2_3);
+
+        StreamReadOptions blockOption = StreamReadOptions.builder().build();
```

Comment:
Why `blockOption`? It does nothing.
BTW you need to test this with blocking too